test/src/blockMonitor/blockchainMachine.test.ts

```diff
@@ -50,34 +55,148 @@ type TestAction = {
     newState: ExampleState;
 };
 
+let componentsCounter = 0;
 class ExampleComponent extends Component<ExampleState, IBlockStub, TestAction> {
+    // make sure each component has a different name
+    public readonly name = "example" + (++componentsCounter); //prettier-ignore
     public async applyAction(actions: TestAction) {}
     public detectChanges(prevState: ExampleState, state: ExampleState): TestAction[] {
         return [{ prevState: prevState, newState: state }];
     }
 }
 
+interface CanEmitAsNewHead {
+    emitAsNewHead(head: IBlockStub): Promise<void>;
+}
+
+class MockBlockProcessor {
+    private newHeadListeners: ((head: Readonly<IBlockStub>) => Promise<void>)[] = [];
+
+    public addNewHeadListener(listener: (head: Readonly<IBlockStub>) => Promise<void>) {
+        this.newHeadListeners.push(listener);
+    }
+    public removeNewHeadListener(listener: (head: Readonly<IBlockStub>) => Promise<void>) {
+        const idx = this.newHeadListeners.findIndex(l => l === listener);
+        if (idx === -1) throw new ApplicationError("No such listener exists.");
+
+        this.newHeadListeners.splice(idx, 1);
+    }
+
+    // Fake the emission of a block (calling all the new_head listeners)
+    public async emitAsNewHead(head: IBlockStub) {
+        return Promise.all(this.newHeadListeners.map(listener => listener(head)));
+    }
+}
+
+describe("ActionStore", () => {
+    let actionStore: ActionStore;
+    let db: any;
+
+    const componentName = "awesome-component";
+    const testActions = [
+        {
+            name: "action1"
+        },
+        {
+            name: "action2"
+        }
+    ];
+
+    beforeEach(async () => {
+        db = LevelUp(EncodingDown<string, any>(MemDown(), { valueEncoding: "json" }));
+        actionStore = new ActionStore(db);
+        await actionStore.start();
```

Comment:
I guess it doesnt really matter, cos our current implementation doesnt do anything. But it might be best to call stop() in afterEach()

Reply:
Done in 6845412, with a guard to prevent stopping the servicecs twice, as some tests already stop some services prematurely.